src/main/webapp/app/exam/participate/exam-navigation-bar/exam-navigation-bar.component.ts

```diff
@@ -134,23 +134,24 @@ export class ExamNavigationBarComponent implements OnInit {
 
     /**
      * Save the currently active exercise and go to the next exercise.
+     * If it is already in the last exercise or if changeExercise is false,
+     * this method does not go to the next exercise but still saves the current one.
      * @param changeExercise whether to go to the next exercise {boolean}
      */
     saveExercise(changeExercise = true) {
-        const newIndex = this.exerciseIndex + 1;
         const submission = ExamParticipationService.getSubmissionForExercise(this.exercises[this.exerciseIndex]);
         // we do not submit programming exercises on a save
         if (submission && this.exercises[this.exerciseIndex].type !== ExerciseType.PROGRAMMING) {
             submission.submitted = true;
         }
-        if (changeExercise) {
-            if (newIndex > this.exercises.length - 1) {
-                // we are in the last exercise, if out of range "change" active exercise to current in order to trigger a save
-                this.changePage(false, this.exerciseIndex, true);
-            } else {
-                this.changePage(false, newIndex, true);
-            }
+
+        // If changeExercise is false, set newIndex to the current index to trigger a save without navigating to a new exercise.
+        let newIndex = changeExercise ? this.exerciseIndex + 1 : this.exerciseIndex;
+        if (newIndex > this.exercises.length - 1) {
```

Comment:
Just a suggestion but I think this would be easier to understand if it was simplified to a single if.

Reply:
Thanks, [previously](https://github.com/ls1intum/Artemis/pull/5413/commits/8a0bd28f972371cb6df36a0894d1a380e8fb54c7) it was similar to your suggestion before but we discussed with @krusche and this way seemed more understandable.